src/components/wasteCalendar/WasteReminderSettings.tsx

```diff
@@ -0,0 +1,361 @@
+  const [error, setError] = useState(false);
+  const [showDatePicker, setShowDatePicker] = useState(false);
+
+  const streetName = locationData.street;
+
+  const { onDayBefore, reminderTime } = state;
+
+  // this state stores the value of the DateTimePicker
+  const [localSelectedTime, setLocalSelectedTime] = useState<Date>(new Date());
+
+  const loadStoredSettings = useCallback(
+    async (street: string) => {
+      setLoading(true);
+      setError(false);
+      const storedSettings = await getReminderSettings();
+
+      if (!areValidReminderSettings(storedSettings)) {
+        setError(true);
+        setLoading(false);
+      } else {
+        const newSettings: ReminderSettings = parseReminderSettings(storedSettings, street);
+
+        // TODO: fix label for stored settings of inactive types ans show them as well
+        // initialize uninitialized types with false
+        Object.keys(types).forEach((key) => {
+          newSettings.activeTypes[key] ??= { active: false };
+        });
+
+        dispatch({ type: ReminderSettingsActionType.OVERWRITE, payload: newSettings });
+        setLocalSelectedTime(newSettings.reminderTime);
+
+        setLoading(false);
+      }
+    },
+    [dispatch, setError, setLoading, types]
+  );
+
+  const onAbortIOS = useCallback(() => {
+    setShowDatePicker(false);
+    setLocalSelectedTime(state.reminderTime);
+  }, [state.reminderTime, setLocalSelectedTime, setShowDatePicker]);
+
+  const onAcceptIOS = useCallback(() => {
+    setShowDatePicker(false);
+    dispatch({ type: ReminderSettingsActionType.UPDATE_TIME, payload: localSelectedTime });
+  }, [localSelectedTime, setShowDatePicker, dispatch]);
+
+  const onDatePickerChange = useCallback(
+    (_, newTime?: Date) => {
+      if (device.platform === 'android') {
+        setShowDatePicker(false);
+
+        if (newTime) {
+          newTime.setMilliseconds(0);
+          newTime.setSeconds(0);
+          dispatch({ type: ReminderSettingsActionType.UPDATE_TIME, payload: newTime });
+        }
+      }
+
+      setLocalSelectedTime((time) => newTime || time);
+    },
+    [setLocalSelectedTime]
+  );
+
+  const onPressDayBefore = useCallback(
+    () => dispatch({ type: ReminderSettingsActionType.UPDATE_ON_DAY_BEFORE, payload: true }),
+    [dispatch]
+  );
+
+  const onPressDayOfCollection = useCallback(
+    () => dispatch({ type: ReminderSettingsActionType.UPDATE_ON_DAY_BEFORE, payload: false }),
+    [dispatch]
+  );
+
+  const updateSettings = useCallback(async () => {
+    const newState = { ...state };
+    let errorOccured = false;
+
+    await Promise.all(
+      Object.keys(state.activeTypes).map(async (typeKey) => {
+        const entry = state.activeTypes[typeKey];
+        // delete inactive entries
+        if (!entry.active && entry.storeId) {
+          const success = await deleteReminderSetting(entry.storeId);
+
+          errorOccured ||= !success;
+        }
+
+        if (entry.active) {
```

Comment:
better optional chain `.active` here, because entry could be nullish?

Reply:
same as above